File: 274. H-Index new.py
class Solution(object):
    def hIndex(self, citations):
        """
        :type citations: List[int]
        :rtype: int
        """
        n = len(citations)
        citations = sorted(citations)
        l = 0
        r = n
        while l < r:
            mid = (l+r) >> 1
            if n-mid <= citations[mid]:
                r = mid
            else:
                l = mid + 1
        return n-l

File: 274. H-Index new.js
/**
 * @param {number[]} citations
 * @return {number}
 */
var hIndex = function(citations) {
    n = citations.length;
    citations.sort(function(a,b){return a-b});
    var l = 0;
    var r = n;
    while (l < r)
    {
        var mid = Math.floor((l+r)/2);
        if (citations[mid] >= n-mid)
            r = mid;
        else
            l = mid + 1;
    }
    return n-l;
};
